diff --git a/models.py b/models.py
index 9ac7906..5da4979 100644
--- a/models.py
+++ b/models.py
@@ -321,10 +321,14 @@ class RegistroTreino(db.Model):
         db.Index('idx_registro_periodo_semana', 'periodo', 'semana'),
     )
 
-    @property
+    @hybrid_property
     def exercicio_id(self):
         return self.exercicio_usuario_id or self.exercicio_base_id
 
+    @exercicio_id.expression
+    def exercicio_id(cls):
+        return db.func.coalesce(cls.exercicio_usuario_id, cls.exercicio_base_id)
+
 
 class HistoricoTreino(db.Model):
     __tablename__ = 'historico_treino'